### ISSUE TITLE
no-variable-construct-id: false positive for function expressions outside Construct classes

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#546]
### Describe the bug

The docs for `no-variable-construct-id` state "Functions outside Construct classes are not flagged", and `shouldSkipIdValidation` accordingly skips `FunctionDeclaration` and `ArrowFunctionExpression` ancestors — but not `FunctionExpression`. A helper written as a function *expression* is flagged while the semantically identical function *declaration* (or arrow function) is not.

**Repro:**

```ts
import { Bucket } from "aws-cdk-lib/aws-s3";
import { Construct } from "constructs";

// function declaration -> valid (documented behavior)
export function createBucketDecl(scope: Construct, id: string): Bucket {
  return new Bucket(scope, id);
}

// function expression -> reported (false positive)
export const createBucketExpr = function (scope: Construct, id: string): Bucket {
  return new Bucket(scope, id);
};
```

### Expected Behavior

Neither function is inside a Construct class; neither should be reported.

### Current Behavior

```
repro/no-variable-construct-id/function-expression.ts  19:28  error  Shouldn't use a parameter as a Construct ID  awscdk/no-variable-construct-id
```

(19:28 is the `new Bucket(scope, id)` inside the function expression; the function declaration is correctly not reported. Line 8 in the same output is an in-constructor control.) Oxlint reports the identical set.

### Plugin

eslint-plugin-awscdk, oxlint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.0, oxlint-plugin-awscdk 1.0.0

### Language Version

TypeScript 6.0.3 (eslint-plugin-awscdk) / TypeScript 7 preview via oxlint-tsgolint (oxlint-plugin-awscdk)

### Additional Context

- Verified on `main` (35b889a), rule-isolated configs.
- Root cause: `packages/*/src/rules/no-variable-construct-id.ts` — `shouldSkipIdValidation` has cases for `ForStatement`/`ForInStatement`/`ForOfStatement`/`WhileStatement`/`DoWhileStatement`, non-constructor `MethodDefinition`, `ArrowFunctionExpression`, and `FunctionDeclaration`, but not `FunctionExpression`.
- Docs reference: `docs/rules/no-variable-construct-id.md` ("Functions outside Construct classes are not flagged.").
